### DISCUSSION TITLE
Any Aspire config tips for debug with visual studio & MS Identity & Blazor?

### DISCUSSION THREAD

**Discussion by swegele** [microsoft/aspire#3264] [Q&A] [answered]
Aspire project has WebAPI and Blazor app.  

Since adding the Blazor App to the solution...now Visual Studio (on F5) starts:
1. the dashboard in a browser window (with links to API and Blazor pulled from their `launchSettings.json` files.
2. the blazor app in a browser window (with a random port selected)

I'm assuming it does number 2 so that it can hook up all the handy dandy features like CSS link and hot-reload to the window with the blazor in it.  But the pesky thing is the AppRegistration for MS Entra has the Redirect URI needs a url.  But I don't know that because the port is randomly picked so it's a different URL each time.

Any tips or tricks from you pros out there?

**Comment by RedTahr** [comment, marked as answer]
my workaround for now is to set 
`"launchBrowser": false, `
in the blazor server projects launchSettings.json file.

means I need to click to open the website from the aspire dashboard, but at least the port number is as defined in the launchSettings.json and not random every time.

refer:
[Aspire project with Blazor WASM/server project opens by itself #3145](https://github.com/dotnet/aspire/issues/3145)
[Aspire 8.0.1 -> second https endpoint with random port #4249](https://github.com/dotnet/aspire/issues/4249)
